modules/doctor/repository.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package doctor
+
+import (
+	"context"
+
+	"code.gitea.io/gitea/models/db"
+	user_model "code.gitea.io/gitea/models/user"
+	"code.gitea.io/gitea/modules/log"
+	repo_service "code.gitea.io/gitea/services/repository"
+
+	"xorm.io/builder"
+)
+
+func handleDeleteOrphanedRepos(ctx context.Context, logger log.Logger, autofix bool) error {
+	test := &consistencyCheck{
+		Name:         "Repos with no existing owner",
+		Counter:      countOrphanedRepos,
+		Fixer:        deleteOrphanedRepos,
+		FixedMessage: "Deleted all content related to orphaned repos",
+	}
+	return test.Run(ctx, logger, autofix)
+}
+
+// countOrphanedRepos count repository where user of owner_id do not exist
+func countOrphanedRepos(ctx context.Context) (int64, error) {
+	return db.CountOrphanedObjects(ctx, "repository", "user", "repository.owner_id=user.id")
+}
+
+// deleteOrphanedRepos delete repository where user of owner_id do not exist
+func deleteOrphanedRepos(ctx context.Context) (int64, error) {
+	batchSize := db.MaxBatchInsertSize("repository")
+	e := db.GetEngine(ctx)
+	var deleted int64
+	adminUser := &user_model.User{IsAdmin: true}
+
+	for {
+		var ids []int64
+		if err := e.Table("`repository`").
+			Join("LEFT", "`user`", "repository.owner_id=user.id").
+			Where(builder.IsNull{"`user`.id"}).
+			Select("`repository`.id").Limit(batchSize).Find(&ids); err != nil {
```

Comment:
I think you forget a where for the ID. It looks like you just get the first n repositories of the batch size and ignore the rest.

Reply:
Isn't absence of user.id for _any_ repository is what you are looking for ?
`Where(builder.IsNull{"`user`.id"}).`

Reply:
I mean the Repository ID

Reply:
@JakobDev I utilize the left join to filter out the broken repos

and yes I limit it as it is run in a for loop ... so its intendet

Reply:
I dont get the issue ... it's working as intendet.
could you provide a code diff or better explanation what issue you did discover please :)

Reply:
Was a mistake by me. Sorry. Normally you need to run something like this to get all repos:
```sql
SELECT * FROM repository WHERE id > 0 ORDER BY id LIMIT 100
SELECT * FROM repository WHERE id > 100 ORDER BY id LIMIT 100
SELECT * FROM repository WHERE id > 200 ORDER BY id LIMIT 100
```
but in this case everything what you get with this Query is deleted, so you catchall repos without duplicates.